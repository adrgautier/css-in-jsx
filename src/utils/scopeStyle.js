const scopeStyle = (rawStyle, uniqueId) => `
    .${uniqueId}{
        ${rawStyle}
    }
`;

export default scopeStyle;
