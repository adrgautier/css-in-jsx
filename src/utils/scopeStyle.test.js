import scopeStyle from './scopeStyle';

describe('scopeStyle', () => {
    it('should match snapshot', () => {
        // GIVEN
        const uniqueId = 'abc123';
        const style = `
            color: black;
            .active {
                color: red;
            }
        `;

        // WHEN
        const scopedStyle = scopeStyle(style, uniqueId);

        // THEN
        expect(scopedStyle).toMatchSnapshot();
    });
});