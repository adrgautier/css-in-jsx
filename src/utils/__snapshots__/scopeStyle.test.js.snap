// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scopeStyle should match snapshot 1`] = `
"
    .abc123{
        
            color: black;
            .active {
                color: red;
            }
        
    }
"
`;
